src/main/kotlin/server/Connection.kt

```diff
@@ -30,7 +30,7 @@ class Connection(private val sendText: (String) -> Unit) {
         } ?: run {
             if (text.startsWith(";")) {
                 val code = text.substringAfter(";")
-                sendText(Compiler.eval(Context(this), code))
+                sendText(Compiler.eval(Context(this), "return $code"))
```

Comment:
After cleaning up stack handling, this was needed to let us see return values of eval expressions.